File: libCompiler/src/main/java/com/duy/pascal/backend/linenumber/LineInfo.java
package com.duy.pascal.backend.linenumber;

public class LineInfo {
    public int line;
    public int column = 0;
    private String sourceFile;

    public LineInfo(int line, String sourceFile) {
        this.line = line;
        this.sourceFile = sourceFile;
    }

    public LineInfo(int line, int column, String sourceFile) {
        this.line = line;
        this.column = column;
        this.sourceFile = sourceFile;
    }

    public int getLine() {
        return line;
    }

    public void setLine(int line) {
        this.line = line;
    }

    public int getColumn() {
        return column;
    }


    public String getSourceFile() {
        return sourceFile;
    }

    @Override
    public String toString() {
        return "Line " + line + (column >= 0 ? ":" + column : "") + " " + sourceFile;

    }

}

File: libCompiler/src/main/java/com/duy/pascal/backend/linenumber/LineInfo.kt
package com.duy.pascal.backend.linenumber

open class LineInfo {
    /**
     * lineInfo in code
     */
    var line: Int = 0

    var column = 0

    /**
     * length of token
     */
    var length = -1

    /**
     * name of source file
     */
    var sourceFile: String? = null
        private set

    constructor(line: Int, sourceFile: String) {
        this.line = line
        this.sourceFile = sourceFile
    }

    constructor(line: Int, column: Int, sourceFile: String) {
        this.line = line
        this.column = column
        this.sourceFile = sourceFile
    }

    constructor(line: Int, column: Int, length: Int, sourceFile: String) {
        this.line = line
        this.column = column
        this.sourceFile = sourceFile
        this.length = length;
    }

    override fun toString(): String {
        return "Line " + line + (if (column >= 0) ":" + column else "") + " " + sourceFile

    }

}
